pages/latest/[urlTitle].tsx

```diff
@@ -63,20 +62,21 @@ const UrlTitle = ({ data, source, toc, error }: PageData): JSX.Element => {
         []
     );
     useEffect(() => {
-        const targets = document.querySelectorAll('h2,h3');
-        if (observer.current) {
-            observer.current.disconnect();
-        }
-        observer.current = new IntersectionObserver(([entry]) => {
-            if (entry.isIntersecting) {
-                setTocActive(entry.target.id);
-            }
-        }, options);
-        const { current: currentObserver } = observer;
-        targets.forEach((target) => currentObserver.observe(target));
-        return () => currentObserver.disconnect();
+        const timer = setTimeout(() => {
+            const observer = new IntersectionObserver((entries) => {
+                entries.forEach((entry) => {
+                    if (entry.isIntersecting) {
+                        setTocActive(entry.target.id);
+                    }
+                });
+            }, options);
+            const targets = document.querySelectorAll('h2,h3');
+            targets.forEach((target) => observer.observe(target));
+            return () => observer.disconnect();
+        }, 1000);
+        return () => clearTimeout(timer);
         // eslint-disable-next-line react-hooks/exhaustive-deps
-    }, [observer.current, options]);
+    }, [toc]);
```

Comment:
Why the `toc` here?

Reply:
Any time the table of content changes, we want to oversee the new titles.